### DISCUSSION TITLE
same files name and path  are replaces after upgrading to webpack 5

### DISCUSSION THREAD

**Discussion by cosmin-dev** [webpack/webpack#14589] [General]
# Bug report

**What is the current behavior?**
I have an issue after upgrade to webpack 5, we have internal components as npm packages and services that use those components and we have the following issue: 
1. Package A have the following folder structure: 
  src 
     - theme
        - default-theme.ts
     -  views
         - Component.tsx 
     - index.ts
2. Service has the following folder structure 
  src 
     - theme
        - default-theme.ts
     -  exported
         - ExportedView.tsx 
     - index.ts
     
After we run build in the service, default-theme of  Package A gets replaced by default-theme of the service. 

This does not happen if the 2 files have different paths or different name. 

This happens with all files that have same name and same path.

**What is the expected behavior?**

default-theme from the services should not override the package default-theme. 

**Other relevant information:**
webpack version:  5.60.0
Node.js version: 14
Operating System: Mac
Additional tools:

Current webpack configuration: 

https://gist.github.com/cosmin-dev/fd581a1b62f4f04fd62cf573dd539ea0

**Comment by alexander-akait** [comment]
Please use gist or github repo to provide code, it is not readable and please clarify you problem

**Reply by cosmin-dev** [reply]
Sure, I've added the webpack configuration to gist, tell me if you need more details about the issue, thanks!

**Reply by alexander-akait** [reply]
Can you provide small example, I don't know your structure and I can't answer

**Reply by alexander-akait** [reply]
Also please provide versions or package.json

**Reply by cosmin-dev** [reply]
Sure, I cannot share the actual code but I've tested with with something basic: 
 I have a file called 'test.ts' inside package A that contains the following: 

`export const buttonLabel = 'test'`

and I have a component called Button.ts insides the views folder
 
```
import {buttonLabel} from '../test'

export const Button  = () => {
// the output should be 'test'
   console.log(buttonLabel)
}

```
So package A folder structure will be : 
- src
  - test.ts
  - views
      -Button.ts
   - index.ts
Now I have a service and I want to use Button from package A. Inside that service I have 

A file called 'test.ts' and have the following: 

`export const buttonLabel = 'My custom button label'
`
And a file called  CustomView.ts
```
import {Button} from 'packageA'
  import {buttonLabel} from '../test'

// it should be 'test' but instead is 'My custom button label'
console.log(Button())
// it should be 'My custom button label'
console.log(buttonLabel)
```

So service folder structure will be : 
- src
  - test.ts
  - views
      -CustomView.ts
   - index.ts


so we have 2 files with the same name and path and it seams that ad the end  I will have only the content of the file where the package is installed. 

package.json dependecies 
https://gist.github.com/cosmin-dev/66c91fa5934a35cf9f8b8eda063f06a6

**Reply by alexander-akait** [reply]
Without configuration I can't help, I don't see webpack problems here

**Comment by cosmin-dev** [comment]
Found the issue, it can be closed. We use tsconfig-paths-webpack-plugin which seams that is incompatible with webpack 5
